### ISSUE TITLE
Accessibility issue in tabulator-tables concerned tags  "aria-required-parent" and "aria-required-children"

### ISSUE THREAD

**Issue by gaurangbhatt19** [olifolkerd/tabulator#4151]
**Describe the bug**
Table component of tabulator is missing required aria-required-children and aria-required-parent
 for violations refer https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships

**Tabulator Info**
v5.4.3

**Working Example**
https://codesandbox.io/s/wandering-silence-6gltpx?file=/src/TabulatorTable.tsx : CodeSandbox

**To Reproduce**
A step by step guide to recreate the issue in your JS Fiddle or Codepen:
1. Go to 'https://codesandbox.io/s/wandering-silence-6gltpx?file=/src/TabulatorTable.tsx '.
2. Run an accessibility tools such as [Accessibility Insights](https://accessibilityinsights.io/) or Axe DevTools .
3. Run the FastPass test from extension and check automated checks
4. Verify that you receive these issues
![image](https://user-images.githubusercontent.com/44087367/228893614-61e86486-4ada-4945-ba1e-cd7042e3081a.png)

**Expected behavior**
Elements with an ARIA role that require child roles contain them and elements with an ARIA role that require parent roles are contained by them

Following elements requires changes

```
<div id="example-table" class="tabulator" role="grid" tabulator-layout="fitColumns" style="height: 210px;">
<div class="tabulator-header" role="rowgroup">
<div class="tabulator-header-contents" role="rowgroup">

```

Recommended solution by accessibility tester software

For elements
```
<div id="example-table" class="tabulator" role="grid" tabulator-layout="fitColumns" style="height: 210px;">
<div class="tabulator-header" role="rowgroup">
```

Fix ONE of the following:
Element has children which are not allowed (see Related paths)
Element has no aria-busy="true" attribute

For
 ```<div class="tabulator-header-contents" role="rowgroup">```

Fix the following:
Required ARIA parents role not present: grid, table, treegrid

**Screenshots**
![image](https://user-images.githubusercontent.com/44087367/228898033-b10cb97d-069d-40f6-b029-c39e2c6dce6a.png)

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Browser : Chrome
 - Version: 111.0.5563.147

**Comment by dzwarg** [comment]
Hello,

This can be resolved by removing the outer div's "rowgroup" role:

Original:
```
<div class="tabulator-header" role="rowgroup">
<div class="tabulator-header-contents" role="rowgroup">
```

Fixed:
```
<div class="tabulator-header">
<div class="tabulator-header-contents" role="rowgroup">
```

Or, collapse these into a single element.

**Comment by dzwarg** [comment]
Here is the line where the extraneous "rowgroup" is added:

https://github.com/olifolkerd/tabulator/blob/0a75da61b24625015483e1f2a87f1f224d1478b2/src/js/core/ColumnManager.js#L93

**Comment by olifolkerd** [comment]
Hey @dzwarg 

Thanks for the information.

Im afraid it isnt as simple as that, removing those simply triggers another error that says those elements need a role.

Tabulator uses more elements  than a standard table, so mapping them directly is proving to be a challenge.

Every time i update the roles to make lighthouse or other assessment tools happy, the rules seem to change and I have to adjust them again next month.

If someone is willing to look at the table structure and provide a conclusive list of all the roles every element should have, i would be more than happy to implement it.

Cheers

Oli :)

**Comment by dineshdevlpr** [comment]
<img width="644" alt="image" src="https://github.com/olifolkerd/tabulator/assets/76520994/ff50d5a0-7a63-41a1-9744-bb9113b35ec5">

Potential Problem:
i) Here a div with class="tabulator-header" have role="rowgroup" and its child div with class="tabulator-header-contents" also have role="rowgroup".

ii) Hence both parent div and its child div are having same role, accessibility error will be there.

Potential Solution:
i) Remove or provide some other appropriate role instead of "rowgroup" to either of the div elem. Ideally the role shouldnt be same. Thats all.

**Comment by olifolkerd** [comment]
@dzwarg yes, but because Tabulator needs to have all of these different tags which are more than a standard table there is no appropriate role to give them, hence the issue
